File: querydsl/src/main/java/com/querydsl/domain/entity/Member.java
package com.querydsl.domain.entity;


import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Builder
@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(indexes = {@Index(name = "idx_name",columnList = "name")})
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "zone_id")
    private Zone zone;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @OneToMany(mappedBy = "member")
    @Builder.Default
    private List<MemberCategory> memberCategoryList = new ArrayList<>();

    private String name;

    private int age;



}

File: webflux-coroutine/client/src/main/kotlin/com/example/client/dao/Member.kt
package com.example.server.dao

import java.time.LocalDateTime

data class Member(
    val id: Long? = null,
    var name: String,
    var teamId: Long? = null,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val modifiedAt: LocalDateTime = LocalDateTime.now()
) {

    // ManyToOne
    var team: Team? = null
}
